Rubberduck.VBEditor.VBA/SafeComWrappers/Application/AccessApp.cs

```diff
@@ -84,6 +96,71 @@ public override IEnumerable<HostDocument> GetDocuments()
             return list;
         }
 
+        public override bool CanOpenDocumentDesigner(QualifiedModuleName moduleName)
+        {
+            return TryGetDocument(moduleName, out _);
+        }
+
+        public override bool TryOpenDocumentDesigner(QualifiedModuleName moduleName)
+        {
+            try
+            {
+                if (moduleName.ComponentName.StartsWith(FormNamePrefix))
+                {
+                    var name = moduleName.ComponentName.Substring(FormNamePrefix.Length);
+                    using (var currentProject = new SafeIDispatchWrapper<_CurrentProject>(Application.CurrentProject))
+                    using (var allForms = new SafeIDispatchWrapper<AllObjects>(currentProject.Target.AllForms))
+                    using (var accessObject = new SafeIDispatchWrapper<AccessObject>(allForms.Target[name]))
+                    using (var doCmd = new SafeIDispatchWrapper<DoCmd>(Application.DoCmd))
+                    {
+                        if (accessObject.Target.IsLoaded &&
+                            accessObject.Target.CurrentView != AcCurrentView.acCurViewDesign)
+                        {
```

Comment:
Can't `Target` return `null`?  This looks like it needs a whole bunch of `IsWrappingNullReference` checks or null propagation.

Reply:
Hmmm. I'm 99.9% sure that Access OM will not return a `null` for an `AccessObject` object; if it doesn't exist or something, it will throw an exception. In fact, if I got a `null` here, I would think it a bug with the `SafeIDispatchWrapper` first. I can see wrapping a `try` around the `allForms.Target[name]` -- that *could* throw if we can't find a matching form. 

That said, I'm hard-pressed to justify a case of adding null handling in this block. This is already quite verbose due to all the extra handling, individually taking references to each intermediate objects and so forth. I'm open to suggestions about we can handle unexpected NREs.

Reply:
Hmmm... That's a tough one.  Now that I'm giving it a closer read, I'm not sure there's much more we could do in terms of recovery other than returning false anyway.  Is the collection access with `allForms.Target[name]` guaranteed to return a non-null result? The main reason I'm skittish is that Excel and Word have no problem handing out nulls.